pydeseq2/dds.py

```diff
@@ -220,6 +220,39 @@ def __init__(
             raise ValueError("NaNs are not allowed in the design factors.")
         self.obs[self.design_factors] = self.obs[self.design_factors].astype(str)
 
+        # Check that design factors don't contain underscores. If so, convert them to
+        # hyphens.
+        if np.any(["_" in factor for factor in self.design_factors]):
+            warnings.warn(
+                """Same factor names in the design contain underscores ('_'). They will
+                be converted to hyphens ('-').""",
+                UserWarning,
+                stacklevel=2,
+            )
+
+            new_factors = [factor.replace("_", "-") for factor in self.design_factors]
+
+            self.obs.rename(
+                columns={
+                    old_factor: new_factor
+                    for (old_factor, new_factor) in zip(self.design_factors, new_factors)
+                },
+                inplace=True,
+            )
+
+            self.design_factors = new_factors
+
+            # Also check continuous factors
+            if self.continuous_factors is not None:
+                self.continuous_factors = [
+                    factor.replace("_", "-") for factor in self.continuous_factors
```

Comment:
It seems like you are doing the same thing for multiple variables. Perhaps you could write a helper function where you just pass those variables, make the switch and return it back instead of repeating the same each time?

Reply:
Indeed! I've added a `replace_underscores` function in `utils.py`.